### ISSUE TITLE
IpSec conn is not working between Rhel7.9 and Rhel8.6 on IKev2

### ISSUE THREAD

**Issue by akki-9922** [libreswan/libreswan#1723]
Hi Team,
I have a base setup on RHEL7.9 for IPsec with said config and My IPsec was working fine.

Host1 is for FrontAPP:
nodedb.conf::
  conn nodedb
        leftid=%fromcert
        left=<nodeapp_Ip>
        leftrsasigkey=%cert
        leftcert="app"
        rightid=%fromcert
        right=<nodedb_IP>
        rightrsasigkey=%cert
        auto=start

Host2 is for DB:
nodeapp.conf::
  conn nodeapp
        leftid=%fromcert
        left=<nodeadb_Ip>
        leftrsasigkey=%cert
        leftcert="db"
        rightid=%fromcert
        right=<nodeapp_IP>
        rightrsasigkey=%cert
        auto=start

I have updated APP server to Rhel8 and after that I can see my connection is not working. and getting below error:
Error from node APP[Updated to R8]:
Received unauthenticated INVALID_KE_PAYLOAD response to DH DH19; resending with suggested DH MODP2048
initial Main Mode message received but no connection has been authorized with policy RSASIG

Errror from Node DB [Still on R7.9]:
 initiator guessed wrong keying material group (ECP_256); responding with INVALID_KE_PAYLOAD requesting MODP2048

In Rhel7.9, deafult is IEKv1 on which connection was working fine but in Rhel8 default is IKEv2 so upgraded node is using IKEv2 by default. 

I want to restore my IPsec and using IKEv2 now.

Can anyone please suggest? I tried multiple ways to setup IKE and ESPin config files but no luck.

**Comment by letoams** [comment]
Add ikev2=no to the updated end Sent using a virtual keyboard on a phoneOn May 30, 2024, at 13:04, Akash Mishra ***@***.***> wrote:﻿
Hi Team,
I have a base setup on RHEL7.9 for IPsec with said config and My IPsec was working fine.
Host1 is for FrontAPP:
nodedb.conf::
conn nodedb
leftid=%fromcert
left=<nodeapp_Ip>
leftrsasigkey=%cert
leftcert="app"
rightid=%fromcert
right=<nodedb_IP>
rightrsasigkey=%cert
auto=start
Host2 is for DB:
nodeapp.conf::
conn nodeapp
leftid=%fromcert
left=<nodeadb_Ip>
leftrsasigkey=%cert
leftcert="db"
rightid=%fromcert
right=<nodeapp_IP>
rightrsasigkey=%cert
auto=start
I have updated APP server to Rhel8 and after that I can see my connection is not working. and getting below error:
Error from node APP[Updated to R8]:
Received unauthenticated INVALID_KE_PAYLOAD response to DH DH19; resending with suggested DH MODP2048
initial Main Mode message received but no connection has been authorized with policy RSASIG
Errror from Node DB [Still on R7.9]:
initiator guessed wrong keying material group (ECP_256); responding with INVALID_KE_PAYLOAD requesting MODP2048
In Rhel7.9, deafult is IEKv1 on which connection was working fine but in Rhel8 default is IKEv2 so upgraded node is using IKEv2 by default.
I want to restore my IPsec and using IKEv2 now.
Can anyone please suggest? I tried multiple ways to setup IKE and ESPin config files but no luck.

**Comment by akki-9922** [comment]
Thanks @letoams  

 ikev2=no is not working in Rhel8.6. 

Redhat given some WA to disable IKEv2 using on both hosts.

   ikev2=never
   ike=aes-sha2,aes-sha1;modp2048
    esp=aes-sha2,aes-sha1

Documentation.  https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/securing_networks/configuring-a-vpn-with-ipsec_securing-networks  

But this is disabling IKEv2. Since I already updated my NodeAPP to Rhel8 so I don't want to use weaker ciphers and security risks.
That is why I am trying to bring up the connection with IKEv2 default in Rhel8.

I am further checking with documents to ensure ipsec working with IKEv2 with Rhel7.9.

**Comment by letoams** [comment]
You can use ikev2=insist on both ends tooSent using a virtual keyboard on a phoneOn May 30, 2024, at 14:11, Akash Mishra ***@***.***> wrote:﻿
Thanks @letoams
ikev2=no is not working in Rhel8.6.
Redhat given some WA to disable IKEv2 using on both hosts.
ikev2=never
ike=aes-sha2,aes-sha1;modp2048
esp=aes-sha2,aes-sha1
Documentation.  https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/securing_networks/configuring-a-vpn-with-ipsec_securing-networks
But this is disabling IKEv2. Since I already updated my NodeAPP to Rhel8 so I don't want to use weaker ciphers and security risks.
That is why I am trying to bring up the connection with IKEv2 default in Rhel8.
I am further checking with documents to ensure ipsec working with IKEv2 with Rhel7.9.

**Comment by akki-9922** [comment]
This I tried but not working. 

Error on Host running with Rhel7.9 , 
initiator guessed wrong keying material group (ECP_256); responding with INVALID_KE_PAYLOAD requesting MODP2048

Error on Host running with Rhel8
initial Main Mode message received but no connection has been authorized with policy RSASIG

I can see that there is a change in code for **AUTHBY_DEFAULTS**.    It seems default is updated to rsasig and ecsda.

**https://github.com/libreswan/libreswan/blame/9b6567328c481cb01605937cbd890e0e7c9ae3b9/include/authby.h#L44**

This could be the reason that ipsec is not working with default parameters used in Rhel7.9 and need to modify due to this change.

Rhel8 - libreswan-4.5-1.el8_x.1.x86_64
Rhel7.9 libreswan-3.25-9.1.el7_9.x86_64

Can u help to check what was the default value used before this change in R7.9 package? 

In Meantime, I am reading trace/logs to get the value exact value in my node.

**Comment by letoams** [comment]
On May 30, 2024, at 15:11, Akash Mishra ***@***.***> wrote:
> 
> ﻿
> This I tried but not working.
> 
> Error on Host running with Rhel7.9 ,
> initiator guessed wrong keying material group (ECP_256); responding with INVALID_KE_PAYLOAD requesting MODP2048
> 

This is not an error but a warning

> Error on Host running with Rhel8
> initial Main Mode message received but no connection has been authorized with policy RSASIG
> 

This means it used ikev1, not ikev2. Make sure both sides use the same ike= setting

**Comment by akki-9922** [comment]
From Logs, it seems it is IKEv2 but still I can see error ::   initial Main Mode message received but no connection has been authorized with policy RSASIG

If we use X.509 certificates , what is default-implicit-authby, is it rsasig? 

IPSec status at NodeAPP[R8]:

000 "nodeapp":   IKEv2 algorithm newest: AES_GCM_16_256-HMAC_SHA2_512-MODP2048
000 "nodeapp":   ESP algorithms: AES_GCM_16_256-NONE, CHACHA20_POLY1305-NONE, AES_CBC_256-HMAC_SHA2_512_256+HMAC_SHA1_96+HMAC_SHA2_256_128, AES_GCM_16_128-NONE, AES_CBC_128-HMAC_SHA1_96+HMAC_SHA2_256_128
000 "`":   ESP algorithm newest: AES_GCM_16_256-NONE; pfsgroup=<Phase1>
000
000 Total IPsec connections: loaded 1, active 1
000
000 State Information: DDoS cookies not required, Accepting new IKE connections
000 IKE SAs: total(1), half-open(0), open(0), authenticated(1), anonymous(0)
000 IPsec SAs: total(1), authenticated(1), anonymous(0)
000
000 #1: "nodeapp":500 STATE_V2_ESTABLISHED_IKE_SA (established IKE SA); REKEY in 28014s; newest ISAKMP; idle;
000 #2: "nodeapp":500 STATE_V2_ESTABLISHED_CHILD_SA (established Child SA); REKEY in 27773s; newest IPSEC; eroute owner; isakmp#1; idle;
000 #2: "nodeapp" esp.89ee8397@xx.xx.xx.xx esp.fa5f221b@xx.xx.xx.xx tun.0@xx.xx.xx.xx tun.0@xx.xx.xx.xx Traffic: ESPin=0B ESPout=0B ESPmax=0B
000
000 Bare Shunt list:
000

IPSec status at nodeDB[R7]::

000 "nodedb":   dpd: action:hold; delay:0; timeout:0; nat-t: encaps:auto; nat_keepalive:yes; ikev1_natt:both
000 "nodedb":   newest ISAKMP SA: #4; newest IPsec SA: #5;
000 "nodedb":   IKEv2 algorithm newest: AES_GCM_16_256-HMAC_SHA2_512-MODP2048
000 "nodedb":   ESP algorithm newest: AES_GCM_16_256-NONE; pfsgroup=<Phase1>
000
000 Total IPsec connections: loaded 2, active 2
000
000 State Information: DDoS cookies not required, Accepting new IKE connections
000 IKE SAs: total(3), half-open(1), open(0), authenticated(2), anonymous(0)
000 IPsec SAs: total(2), authenticated(2), anonymous(0)
000
000 #4: "nodedb":500 STATE_PARENT_R2 (received v2I2, PARENT SA established); EVENT_SA_REPLACE in 3049s; newest ISAKMP; idle; import:respond to stranger
000 #5: "nodedb":500 STATE_V2_IPSEC_R (IPsec SA established); EVENT_SA_REPLACE in 28249s; newest IPSEC; eroute owner; isakmp#4; idle; import:respond to stranger
000 #5: "nodedb" esp.fa5f221b@xx.xx.xx.xx esp.89ee8397@xx.xx.xx.xx tun.0@xx.xx.xx.xx tun.0@xx.xx.xx.xx ref=0 refhim=0 Traffic: ESPin=0B ESPout=0B! ESPmax=0B
000 #9: "nodedb":500 STATE_MAIN_I1 (sent MI1, expecting MR1); EVENT_v1_RETRANSMIT in 4s; nodpd; idle; import:admin initiate

**Comment by letoams** [comment]
On May 30, 2024, at 16:37, Akash Mishra ***@***.***> wrote:
> 
> ﻿
> From Logs, it seems it is IKEv2 but still I can see error :: initial Main Mode message received
> 
The other end is usung ikev1. Main mode is ikev1

**Comment by akki-9922** [comment]
If we have different IKE versions[v1 and v2] on each side, what is the impact besides these messages.  I can see IPsec connection is established with IKEv2 with the above messages. 

What is the recommendation?

**Comment by letoams** [comment]
On Fri, 31 May 2024, Akash Mishra wrote:

> If we have different IKE versions[v1 and v2] on each side, what is the impact besides these messages. I can
> see IPsec connection is established with IKEv2 with the above messages.

It won't work. RHEL changed the meaning of ikev2=yes|no versus upstream
libreswan. I don't remember exactly how they define it. We used to have
yes, no, insist, propose. And a connection could switch between trying
ike versions. This was removed because it didnt work well and no other
vendor implemented it. So ikev2= option is now only yes|no without any
fallback/switching. So if you see Main Mode messages, you are trying
ikev1 and not ikev2 and it will never work.

> What is the recommendation?

Set both sides to either ikev2=no or ikev2=yes - on older RHELs, set
it to "insist" instead of "yes" for ike2. See the man page to see if
your RHEL is "old".
